src/ml/neural_net/tf_activity_classifier_backend.cpp

```diff
@@ -0,0 +1,195 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+*/
+
+#include <ml/neural_net/tf_activity_classifier_backend.hpp>
+
+#include <iostream>
+#include <vector>
+
+#include <ml/neural_net/float_array.hpp>
+#include <ml/neural_net/tf_compute_context.hpp>
+#include <core/util/try_finally.hpp>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <pybind11/eval.h>
+
+namespace turi {
+namespace neural_net {
+
+namespace py = pybind11;
+
+using turi::neural_net::float_array;
+using turi::neural_net::float_array_map;
+using turi::neural_net::shared_float_array;
+
+
+template <typename CallFunc> 
+auto call_pybind_function(CallFunc&& func) -> decltype(func()) {
```

Comment:
Why is the return type `auto`? Can't it just be `void` here? We don't return anything.

I think we should also prefer `const CallFunc&` as the argument type, since all we do with `func` is call it? (https://google.github.io/styleguide/cppguide.html#Rvalue_references)